integrations/integration.php

```diff
@@ -22,14 +36,29 @@ abstract class Integration {
 	 */
 	protected bool $is_active = false;
 
+	/**
+	 * Constructor.
+	 *
+	 * @param string $slug Slug of the integration.
+	 */
+	public function __construct( string $slug ) {
+		$this->slug = $slug;
+	}
+
 	/**
 	 * Activates this integration with an optional configuration value.
 	 *
-	 * @param array  $config An associative array of configuration values for the integration.
+	 * @param array $config An associative array of configuration values for the integration.
+	 *
+	 * @throws LogicException Exception if integration is already activated.
 	 *
 	 * @private
 	 */
 	public function activate( array $config = [] ): void {
+		if ( $this->is_active() ) {
+			throw new LogicException( sprintf( 'VIP Integration with slug "%s" is already activated.', $this->get_slug() ) );
```

Comment:
This can result in fatal errors on customer sites which is not ideal. We may want to handle it a bit more gracefully: log warning and skip the double-activation request.

Reply:
Addressed in https://github.com/Automattic/vip-go-mu-plugins/pull/4660/commits/266b1587da401cef0ce8d60ae702cf630bdf446c (just skipping silently without error_log, let me know if its compulsory)

Reply:
Added `trigger_error`